services/QuillLMS/engines/evidence/app/controllers/evidence/research/gen_ai/trials_controller.rb

```diff
@@ -6,84 +6,57 @@ module GenAI
       class TrialsController < ApplicationController
         include TrialsHelper
 
-        def index
-          @page = params[:page].to_i > 0 ? params[:page].to_i : 1
-          @per_page = 50
-          @num_trials = Trial.count
-          @trials = Trial.all.order(id: :desc).offset((@page - 1) * @per_page).limit(@per_page)
-        end
-
         def new
-          @trial = Trial.new
+          @trial = dataset.trials.new
           @llms = LLM.all
-          @stem_vaults = StemVault.all
           @llm_prompt_templates = LLMPromptTemplate.all
           @g_evals = GEval.selectable
+          @guidelines = dataset.stem_vault.guidelines
```

Comment:
Could stem_vault be nil here? Might be worth doing a `nil` audit for these nested relationships.

Reply:
dataset has a `belongs_to  :stem_vault`, so this one should not require safe navigation.